kms-message/src/kms_kmip_response.c

```diff
@@ -0,0 +1,320 @@
+#include "kms_message/kms_kmip_response.h"
+
+#include "kms_message_private.h"
+#include "kms_kmip_reader_writer_private.h"
+#include "kms_kmip_result_reason_private.h"
+#include "kms_kmip_result_status_private.h"
+
+#include <stdlib.h>
+#include <inttypes.h>
+
+static bool
+check_and_require_kmip (kms_response_t *res)
+{
+   if (res->provider != KMS_REQUEST_PROVIDER_KMIP) {
+      KMS_ERROR (res, "Function requires KMIP request");
+      return false;
+   }
+   return true;
+}
+
+/*
+Example of an error message:
+<ResponseMessage tag="0x42007b" type="Structure">
+ <ResponseHeader tag="0x42007a" type="Structure">
+  <ProtocolVersion tag="0x420069" type="Structure">
+   <ProtocolVersionMajor tag="0x42006a" type="Integer" value="1"/>
+   <ProtocolVersionMinor tag="0x42006b" type="Integer" value="0"/>
+  </ProtocolVersion>
+  <TimeStamp tag="0x420092" type="DateTime" value="2021-10-01T14:43:13-0500"/>
+  <BatchCount tag="0x42000d" type="Integer" value="1"/>
+ </ResponseHeader>
+ <BatchItem tag="0x42000f" type="Structure">
+  <Operation tag="0x42005c" type="Enumeration" value="10"/>
+  <ResultStatus tag="0x42007f" type="Enumeration" value="1"/>
+  <ResultReason tag="0x42007e" type="Enumeration" value="1"/>
+  <ResultMessage tag="0x42007d" type="TextString" value="ResultReasonItemNotFound"/>
+ </BatchItem>
+</ResponseMessage>
+*/
+static bool
+kms_kmip_response_ok (kms_response_t *res)
+{
+   kmip_reader_t *reader = NULL;
+   size_t pos;
+   size_t len;
+   uint32_t result_status;
+   uint32_t result_reason = 0;
+   char *result_message = "";
```

Comment:
Does this string need to be freed at the end, or does destroying the reader free it? Also, would this be better initialized to NULL since string literals are const?

Reply:
Destroying the reader frees it. I changed this to `const char*` to clarify.